### DISCUSSION TITLE
CTkScrollableFrame for both axes

### DISCUSSION THREAD

**Discussion by hoffmanlucas** [TomSchimansky/CustomTkinter#1813] [Ideas]
Could we get a scrollable frame with scroll bars for both axes?
It would get the orientation = "both" and allow scroll horizontally and vertically. 

I am trying to achieve this by editing the ctk_scrollable_frame.py, but the bars are not moving. Any thoughts? 

![image](https://github.com/TomSchimansky/CustomTkinter/assets/191785/5a2a9504-e101-47d3-9330-282b2d8e8312)

![image](https://github.com/TomSchimansky/CustomTkinter/assets/191785/24716239-3403-437d-9d0b-09b850ac00ff)

![image](https://github.com/TomSchimansky/CustomTkinter/assets/191785/f2c1b845-d091-4669-bc35-2b8b82ffad5f)

**Comment by Akascape** [comment]
@hoffmanlucas You can use this separate [CTkXYFrame](https://github.com/Akascape/CTkXYFrame) widget which have both horizontal and vertical scrollbars in the same frame. It also dynamically hides the scrollbars besed on the size. 

![Screenshot](https://github.com/TomSchimansky/CustomTkinter/assets/89206401/05298e8d-cc9d-48ca-8b27-9943176d5c1a)

**Reply by JPLost** [reply]
Regarding the CTkXYFrame, it does not seem to properly support canvas with autosize.
        
        self.middle_frame = CTkXYFrame(self)
        self.middle_frame.grid(row=0, column=1, padx=2, pady=2, sticky="nesw")
        self.canvas = customtkinter.CTkCanvas(master=self.middle_frame)
        self.canvas.pack(expand=True, fill="both")

The idea is that the canvas fills whole middle frame by default and when image is loaded it recognizes the 1:1 image and adjusts scrolls for that. I'm also thinking about scaling function? 

15.7
After considering my thinking, I can override few places in class myself.

**Comment by trallallero** [comment]
[ctk_scrollable_frame2.py](https://github.com/user-attachments/files/22118759/ctk_scrollable_frame2.py)

For those interested in having a both vertical and horizontal scrollframe, here it is (ctk_scrollable_frame2.py). Was quite easy. Just need to add the imports in the files to include it in the project (and maybe rename it to a better name)
